src/egon/data/datasets/electricity_demand_timeseries/cts_buildings.py

```diff
@@ -1542,10 +1540,70 @@ def get_cts_heat_peak_load():
         write_table_to_postgres(
             df_peak_load,
             BuildingHeatPeakLoads,
-            engine=engine,
             drop=False,
             index=False,
             if_exists="append",
         )
 
         log.info(f"Peak load for {scenario} exported to DB!")
+
+
+def assign_voltage_level_to_buildings():
+    """
+    Add voltage level to all buildings by summed peak demand.
+
+    All entries with same building id get the voltage level corresponding
+    to their summed residential and cts peak demand.
+    """
+
+    with db.session_scope() as session:
+        cells_query = session.query(BuildingElectricityPeakLoads)
+
+        df_peak_loads = pd.read_sql(
+            cells_query.statement,
+            cells_query.session.bind,
+        )
+
+    df_peak_load_buildings = df_peak_loads.groupby("building_id")[
+        "peak_load_in_w"
+    ].sum()
+    df_peak_load_buildings = df_peak_load_buildings.to_frame()
+    df_peak_load_buildings.loc[:, "voltage_level"] = 0
+
+    # Identify voltage_level by thresholds defined in the eGon project
+    df_peak_load_buildings.loc[
+        df_peak_load_buildings["peak_load_in_w"] < 0.1 * 1e6, "voltage_level"
+    ] = 7
+    df_peak_load_buildings.loc[
+        df_peak_load_buildings["peak_load_in_w"] > 0.1 * 1e6, "voltage_level"
+    ] = 6
+    df_peak_load_buildings.loc[
+        df_peak_load_buildings["peak_load_in_w"] > 0.2 * 1e6, "voltage_level"
+    ] = 5
+    df_peak_load_buildings.loc[
+        df_peak_load_buildings["peak_load_in_w"] > 5.5 * 1e6, "voltage_level"
+    ] = 4
+    df_peak_load_buildings.loc[
+        df_peak_load_buildings["peak_load_in_w"] > 20 * 1e6, "voltage_level"
+    ] = 3
+    df_peak_load_buildings.loc[
+        df_peak_load_buildings["peak_load_in_w"] > 120 * 1e6, "voltage_level"
+    ] = 1
```

Comment:
Actually, you did not include the upper limit in level 7 according to the [intervals as defined](https://redmine.iks.cs.ovgu.de/oe/projects/ego-n/wiki/Definition_of_thresholds_for_voltage_level_assignment). I know, others did it the same way but this shouldn't be an excuse :).
I'll fix this.

Reply:
Thx! Sorry for that laziness. Just copy pasted from the other modules.

Reply:
@nesnoj I don't see any further commits. Is this still open?